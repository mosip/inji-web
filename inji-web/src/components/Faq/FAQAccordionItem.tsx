import React from "react";
import {FAQAccordionItemProps} from "../../types/components";
import {IoIosArrowDown, IoIosArrowUp} from "react-icons/io";
import {renderContent} from "../../utils/builder";

export const FAQAccordionItem: React.FC<FAQAccordionItemProps> = (props) => {
    const isOpen = props.id === props.open;

    return (
        <div
            className="w-full rounded-md mb-4 transition-all duration-300 bg-white shadow-[-1px_-0.5px_6px_rgba(0,0,0,0.07),1px_4px_6px_rgba(0,0,0,0.07)]"
            data-testid="Faq-Item-Container"
        >
            <button
                data-testid="Faq-Item-Title-Button"
                className={`w-full px-4 sm:px-6 py-6 sm:py-8 text-left font-semibold text-base sm:text-lg text-iw-title flex justify-between items-center hover:bg-iw-faqAccordionHover ${
                    isOpen ? 'hover:rounded-t-md' : 'hover:rounded-md'
                } focus:outline-none transition-colors`}
                onClick={() => props.setOpen(isOpen ? -1 : props.id)}
            >
                <span data-testid="Faq-Item-Title-Text">{props.title}</span>
                <span className="ml-2 sm:ml-4">
                    {isOpen ? (
                        <IoIosArrowUp
                            size={18}
                            className="sm:size-[20px]"
                            color="#04051D"
                            data-testid="Faq-Item-UpArrow"
                        />
                    ) : (
                        <IoIosArrowDown
                            size={18}
                            className="sm:size-[20px]"
                            color="#04051D"
                            data-testid="Faq-Item-DownArrow"
                        />
                    )}
                </span>
            </button>

            {isOpen && (
                <div className="bg-iw-background rounded-b-md">
                    <div className="px-4 sm:px-6">
                        <div
                            className="border-t border-gray-200 py-5 sm:py-6 text-sm sm:text-base text-iw-title leading-relaxed"
                            data-testid="Faq-Item-Content-Text"
                        >
                            {props.content.map((content) => (
                                <p key={props.key} className="mb-3 last:mb-0">
                                    {renderContent(content)}
                                </p>
                            ))}
                        </div>
                    </div>
                </div>
            )}
        </div>
    );
};
