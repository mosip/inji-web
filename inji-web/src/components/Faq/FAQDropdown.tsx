import React, {useState} from 'react';
import {useNavigate} from 'react-router-dom';
import {useTranslation} from 'react-i18next';
import DropdownArrowIcon from '../Common/DropdownArrowIcon';
import {ROUTES} from "../../utils/constants";

// TODO This component is not being used in the current codebase, but it is kept for future use. We have to check if this can be generalized and used in other places as well.
export const FAQDropdown: React.FC = () => {
    const navigate = useNavigate();
    const [isOpen, setIsOpen] = useState(false);
    const {t} = useTranslation('FAQDropdown');

    return (
        <div
            className={'flex flex-row justify-center items-center'}
            data-testid={'faq-dropdown-outer-div'}
            onBlur={() => setIsOpen(false)}
            tabIndex={0}
            role="button"
        >
            <div className="relative inline-block ms-1">
                <button
                    type="button"
                    className="inline-flex items-center"
                    data-testid={'faq-dropdown-button'}
                    onMouseDown={() => setIsOpen((open) => !isOpen)}
                >
                    <p data-testid={`selected-dropDown`}>{t('heading')}</p>
                    <DropdownArrowIcon isOpen={isOpen} />
                </button>

                {isOpen && (
                    <div className="absolute w-40 right-0 mt-3 rounded-md shadow-lg bg-iw-background overflow-hidden font-normal">
                        <ul className="py-1 divide-y divide-gray-200">
                            <li data-testid={`faq-dropdown-item`}>
                                <button
                                    type="button"
                                    className="w-full px-4 py-2 text-left text-sm font-semibold hover:bg-gray-100 flex items-center justify-between flex-row"
                                    onMouseDown={() => {
                                        setIsOpen((open) => !open);
                                        navigate(ROUTES.FAQ);
                                    }}
                                >
                                    {t('item1')}
                                </button>
                            </li>
                        </ul>
                    </div>
                )}
            </div>
        </div>
    );
};
