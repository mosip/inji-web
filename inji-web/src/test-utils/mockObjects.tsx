import { IssuerConfigurationObject } from '../types/data';
import { LandingPageWrapperProps } from '../components/Common/LandingPageWrapper';
import { NavBarProps } from '../types/components';
export const mockIssuerDisplayArrayObject = {
    name: "Name",
    language: "en",
    locale: "en",
    logo: {
        url: "https://url.com",
        alt_text: "alt text of the url"
    },
    title: "Title",
    description: "Description",
};

export const mockCredentialTypeDisplayArrayObject = {
    name: "Name",
    locale: "en",
    logo: "https://url.com",
};

export const mockCredentials: IssuerConfigurationObject = {
    credentials_supported: [
        {
            name: "InsuranceCredential",
            scope: "mosip_vc_ldp",
            display: [
                {
                    name: "Health Insurance",
                    locale: "en",
                    logo: "https://url.com"
                }
            ]
        },
        {
            name: "AnotherCredential",
            scope: "mosip_ldp_vc",
            display: [
                {
                    name: "Another Credential",
                    locale: "en",
                    logo: "https://url.com"
                }
            ]
        }
    ],
    "authorization_endpoint": "https://env.net/authorize",
    "grant_types_supported": ["authorization_code"]
};


export const mockIssuerObject = {
        name: 'Test Issuer',
        desc: 'Test Description',
        protocol: 'OTP' as 'OTP', // Explicitly set the type to 'OTP' or 'OpenId4VCI'
        issuer_id: 'test-issuer',
        authorization_endpoint: 'https://auth.test.com',
        credentials_endpoint: 'https://credentials.test.com',
        display: [{
            name: "Name",
            language: "en",
            locale: "en",
            logo: {
                url: "https://url.com",
                alt_text: "alt text of the url"
            },
            title: "Title",
            description: "Description",
        }],
        client_id: 'test-client-id',
        redirect_uri:'test-redirect-uri',
        token_endpoint:'test-token_endpoint',
        proxy_token_endpoint:'test-proxy_token_endpoint',
        client_alias:'',
        ovp_qr_enabled: true,
        scopes_supported: ['scope1', 'scope2']
}

export const mockIssuerObjectList =  [
    {
        name: 'Issuer 1',
        desc: 'Description 1',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer1',
        authorization_endpoint: 'https://issuer1.com/auth',
        credentials_endpoint: 'https://issuer1.com/credentials',
        display: [
            {
                name: 'Issuer 1',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer1.com/logo.png', alt_text: 'Issuer 1 Logo' },
                title: 'Issuer 1 Title',
                description: 'Issuer 1 Description',
            },
        ],
        client_id: 'client1',
        redirect_uri: 'https://issuer1.com/redirect',
        token_endpoint: 'https://issuer1.com/token',
        proxy_token_endpoint: 'https://issuer1.com/proxy-token',
        client_alias: 'client1-alias',
        ovp_qr_enabled: true,
        scopes_supported: ['openid', 'profile'],
    },
    {
        name: 'Issuer 2',
        desc: 'Description 2',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer2',
        authorization_endpoint: 'https://issuer2.com/auth',
        credentials_endpoint: 'https://issuer2.com/credentials',
        display: [
            {
                name: 'Issuer 2',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer2.com/logo.png', alt_text: 'Issuer 2 Logo' },
                title: 'Issuer 2 Title',
                description: 'Issuer 2 Description',
            },
        ],
        client_id: 'client2',
        redirect_uri: 'https://issuer2.com/redirect',
        token_endpoint: 'https://issuer2.com/token',
        proxy_token_endpoint: 'https://issuer2.com/proxy-token',
        client_alias: 'client2-alias',
        ovp_qr_enabled: false,
        scopes_supported: ['openid', 'profile'],
    },
]

export const mockLandingPageWrapperProps: LandingPageWrapperProps = {
    icon: <div data-testid="Test-Icon">Icon</div>,
    title: "Test Title",
    subTitle: "Test SubTitle",
    gotoHome: true,
};

export const mockNavBarProps: NavBarProps = {
    title: 'Test Title',
    link: '/test-link',
    search: true,
};