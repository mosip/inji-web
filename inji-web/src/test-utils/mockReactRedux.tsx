jest.mock('react-redux', () => {
    const ActualReactRedux = jest.requireActual('react-redux');
    return {
      ...ActualReactRedux,
      useSelector: jest.fn(),
      useDispatch: jest.fn(),
    };
  });
  
  // Helper to set the mock implementation of useSelector:
  export const setMockUseSelectorState = (state: any) => {
    const { useSelector } = require('react-redux') as { useSelector: jest.Mock };
    useSelector.mockImplementation((selector: any) => selector(state));
  };
  
  export const setMockUseDispatchReturnValue = (dispatchFn: jest.Mock) => {
    const { useDispatch } = require('react-redux') as { useDispatch: jest.Mock };
    (useDispatch as jest.Mock).mockReturnValue(dispatchFn);
};