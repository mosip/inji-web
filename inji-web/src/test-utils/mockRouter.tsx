export const mockNavigateFn = jest.fn();

jest.mock('react-router-dom', () => {
  const Actual = jest.requireActual('react-router-dom');
  return {
    ...Actual,
    useNavigate: () => mockNavigateFn,
    useLocation: jest.fn().mockReturnValue({ pathname: '/' }),
    Outlet: () => <div data-testid="Outlet">Page content</div>,
  };
});

export const setMockUseLocation = (locationObj: { pathname: string }) => {
  const { useLocation } = require('react-router-dom') as { useLocation: jest.Mock };
  useLocation.mockReturnValue(locationObj);
};
