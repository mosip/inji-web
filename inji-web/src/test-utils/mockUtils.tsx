import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, BrowserRouter as Router, Routes } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

// Mock for storage module
export const mockStorageModule = () => {
    jest.mock('../utils/storage.ts', () => ({
        storage: {
            getItem: jest.fn(),
            setItem: jest.fn(),
            SESSION_INFO: 'SESSION_INFO',
            SELECTED_LANGUAGE: 'selectedLanguage'
        }
    }));
};

// Mock for localStorage
export const mockLocalStorage = () => {
    let store: { [key: string]: string } = {};
  
    const localStorageMock = {
      getItem: jest.fn((key: string) => store[key] || null),
      setItem: jest.fn((key: string, value: string) => {
        store[key] = value;
      })
    };
  
    Object.defineProperty(window, 'localStorage', { 
      value: localStorageMock,
      writable: true
    });
  
    return localStorageMock;
  };

// API Mock
export const mockApi = {
    authorizationRedirectionUrl: 'http://mockurl.com'
};

// Crypto Mock
export const mockCrypto = {
    getRandomValues: (array: Uint32Array) => {
        for (let i = 0; i < array.length; i++) {
            array[i] = 4294967296;
        }
        return array;
    },
    subtle: {} as SubtleCrypto,
    randomUUID: () => '123e4567-e89b-12d3-a456-426614174000'
} as Crypto;

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    const mockNavigate = jest.fn();
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: mockNavigate,
    }));
};

export const mockUseSelector = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn(),
    }));
};

export const setMockUseSelectorState = (state: any) => {
    const useSelectorMock = require('react-redux').useSelector;
    useSelectorMock.mockImplementation((selector: any) => selector(state));
};

export const mockUseGetObjectForCurrentLanguage = () => {
    jest.mock('../utils/i18n', () => ({
        getObjectForCurrentLanguage: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

export const mockUseSearchCredentials = () => {
    jest.mock('../components/Credentials/SearchCredential', () => ({
        SearchCredential: () => <></>,
    }));
 };

export const mockUseDispatch = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useDispatch: jest.fn(),
    }));
};
              
export const mockUseSpinningLoader = () => {
    jest.mock('../components/Common/SpinningLoader', () => ({
        SpinningLoader: () => <></>,
    }));
};

export const mockUseLanguageSelector = () => {
    jest.mock('../components/Common/LanguageSelector', () => ({
        LanguageSelector: () => <></>,
    }));
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

export const renderWithProvider = (element: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {element}
            </Router>
        </Provider>,
        options
    );
};
    
export const mockUseParams = ()=>{
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useParams: () => ({ issuerId: 'test-issuer-id' }),
      }));   
};

export const mockUseapiObject = () =>{
    jest.mock('../utils/api', () => ({
        api: {
          mimotoHost: 'http://mocked-api-host',
        },
      }));
};

export const mockUseFetch = () =>{
    jest.mock('../hooks/useFetch', () => {
        const RequestStatus = {
          LOADING: 'LOADING',
          DONE: 'DONE',
          ERROR: 'ERROR',
        };
        return {
          useFetch: () => jest.fn(),
          RequestStatus,
        };
      });
};

export const mockUseToast = () =>{
    jest.mock('react-toastify', () => ({
      toast: {
        error: jest.fn(),
      },
    }));
};

export const mockusemisc = ()=>{
    jest.mock('../utils/misc', () => ({
        downloadCredentialPDF: jest.fn(),
        getErrorObject: jest.fn(),
        getTokenRequestBody: jest.fn(),
      }));
};

export const mockusei18n = ()=>{
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
          t: (key: string) => key,
        }),
        initReactI18next: {
          type: '3rdParty',
          init: jest.fn(),
        },
      }));

};

export const mockWindowLocation = (url: string) => {
    const location = new URL(url) as unknown as Location;
    Object.defineProperty(window, 'location', {
      value: location,
      writable: true,
    });
  };

export const renderWithRouter = (Element: React.ReactElement, { route = '/' } = {}) => {
    window.history.pushState({}, 'Test page', route);
    return render(
      <BrowserRouter>
        <Provider store={reduxStore}>
          <Routes>
            <Route path="*" element={Element} />
          </Routes>
        </Provider>
      </BrowserRouter>
    );
  };

