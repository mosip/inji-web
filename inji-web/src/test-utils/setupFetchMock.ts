export const fetchMock = jest.fn();

beforeAll(() => {
    global.fetch = fetchMock;
});

beforeEach(() => {
    fetchMock.mockReset();
});
