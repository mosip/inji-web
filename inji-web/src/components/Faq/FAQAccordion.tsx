import React, {useState} from "react";
import {FAQAccordionItem} from "./FAQAccordionItem";
import {useTranslation} from "react-i18next";
import { constructContent } from "../../utils/builder";
import {FAQAccordionItemType} from '../../types/data';

export const FAQAccordion: React.FC = () =>{
    
    const [open, setOpen] = useState(0);
    const {t} = useTranslation("FAQ");

    const accordionItems: FAQAccordionItemType[] = [
        {
            key: 'item1',
            title: t('item1.title'),
            content: constructContent([t('item1.description1')], false)
        },
        {
            key: 'item2',
            title: t('item2.title'),
            content: constructContent([t('item2.description1')], false)
        },
        {
            key: 'item3',
            title: t('item3.title'),
            content: constructContent(
                [
                    t('item3.description1'),
                    t('item3.description2'),
                    t('item3.description3')
                ],
                false
            )
        },
        {
            key: 'item4',
            title: t('item4.title'),
            content: constructContent([t('item4.description1')], false)
        },
        {
            key: 'item5',
            title: t('item5.title'),
            content: constructContent([t('item5.description1')], false)
        },
        {
            key: 'item6',
            title: t('item6.title'),
            content: constructContent([t('item6.description1')], false)
        },
        {
            key: 'item7',
            title: t('item7.title'),
            content: constructContent(
                [
                    t('item7.description1'),
                    t('item7.description2'),
                    t('item7.description3'),
                    t('item7.description4')
                ],
                false
            )
        },
        {
            key: 'item8',
            title: t('item8.title'),
            content: constructContent(
                [
                    t('item8.description1'),
                    t('item8.description2'),
                    t('item8.description3'),
                    t('item8.description4'),
                    t('item8.description5')
                ],
                false
            )
        },
        {
            key: 'item9',
            title: t('item9.title'),
            content: constructContent([t('item9.description1')], false)
        },
        {
            key: 'item10',
            title: t('item10.title'),
            content: constructContent([t('item10.description1')], false)
        },
        {
            key: 'item11',
            title: t('item11.title'),
            content: constructContent(
                [
                    t('item11.description1'),
                    t('item11.description2'),
                    t('item11.description3'),
                    t('item11.description4')
                ],
                false
            )
        },
        {
            key: 'item12',
            title: t('item12.title'),
            content: constructContent([t('item12.description1')], false)
        },
        {
            key: 'item13',
            title: t('item13.title'),
            content: constructContent(
                [t('item13.description1'), t('item13.description2')],
                false
            )
        },
        {
            key: 'item14',
            title: t('item14.title'),
            content: constructContent([t('item14.description1')], false)
        },
        {
            key: 'item15',
            title: t('item15.title'),
            content: constructContent([t('item15.description1')], false)
        },
        {
            key: 'item16',
            title: t('item16.title'),
            content: constructContent([t('item16.description1')], false)
        },
        {
            key: 'item17',
            title: t('item17.title'),
            content: constructContent([t('item17.description1')], false)
        },
        {
            key: 'item18',
            title: t('item18.title'),
            content: constructContent(
                [
                    t('item18.description1'),
                    t('item18.description2'),
                    t('item18.description3')
                ],
                false
            )
        },
        {
            key: 'item19',
            title: t('item19.title'),
            content: constructContent([t('item19.description1')], false)
        },
        {
            key: 'item20',
            title: t('item20.title'),
            content: constructContent([t('item20.description1')], false)
        },
        {
            key: 'item21',
            title: t('item21.title'),
            content: constructContent([t('item21.description1')], true)
        },
        {
            key: 'item22',
            title: t('item22.title'),
            content: constructContent([t('item22.description1')], true)
        },
        {
            key: 'item23',
            title: t('item23.title'),
            content: constructContent(
                [
                    t('item23.description1'),
                    t('item23.description2'),
                    t('item23.description3')
                ],
                false
            )
        }
    ];

    return (
        <React.Fragment>
            <div data-testid="Faq-Accordion-Container">
                {accordionItems.map((item, index) => (
                    <FAQAccordionItem
                        id={index}
                        key={item.key}
                        title={item.title}
                        content={item.content}
                        open={open}
                        setOpen={setOpen}
                    />))
                }
            </div>
        </React.Fragment>
    );
};
