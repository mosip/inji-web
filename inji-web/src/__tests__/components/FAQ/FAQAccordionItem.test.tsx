import React from 'react';
import {fireEvent, screen} from '@testing-library/react';
import {FAQAccordionItem} from "../../../components/Faq/FAQAccordionItem";
import { renderWithProvider } from '../../../test-utils/mockUtils';


describe("Testing the Layout of Faq Accordion",() => {
    test('Check if the layout is matching with the snapshots', () => {
        const openHandler = jest.fn();
        const{asFragment} = renderWithProvider(<FAQAccordionItem id={1} title={"FaqTitle"} content={["FaqContent"]} open={1} setOpen={openHandler} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Check if current Faq item is not open, it should not show description', () => {
        const openHandler = jest.fn();
        renderWithProvider(
            <FAQAccordionItem
                id={1}
                title={'FaqTitle'}
                content={['FaqContent']}
                open={0}
                setOpen={openHandler}
            />
        );
        const faqElement = screen.getByTestId('faq-Item-Container');
        expect(faqElement.childElementCount).toBe(1);
    });
    test('Check if current Faq item is open, it should show description', () => {
        const openHandler = jest.fn();
        renderWithProvider(
            <FAQAccordionItem
                id={1}
                title={'FaqTitle'}
                content={['FaqContent']}
                open={1}
                setOpen={openHandler}
            />
        );
        const faqElement = screen.getByTestId('Faq-Item-Container');
        expect(faqElement.childElementCount).toBe(2);
    });

});
describe('Testing the Functionality of Faq Accordition Item', () => {
    test('Check if current Faq item is open, it should show description', () => {
        const openHandler = jest.fn();
        renderWithProvider(
            <FAQAccordionItem
                id={1}
                title={'FaqTitle'}
                content={['FaqContent']}
                open={0}
                setOpen={openHandler}
            />
        );
        const faqElement = screen.getByTestId('Faq-Item-Container');
        expect(faqElement.childElementCount).toBe(1);
        const buttonElement = screen.getByTestId('Faq-Item-Title-Button');
        fireEvent.click(buttonElement);
        expect(openHandler).toHaveBeenCalled();
    });
});
