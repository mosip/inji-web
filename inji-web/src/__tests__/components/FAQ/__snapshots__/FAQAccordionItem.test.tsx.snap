// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of Faq Accordion Check if the layout is matching with the snapshots 1`] = `
<DocumentFragment>
  <div
    class="w-full rounded-md mb-4 transition-all duration-300 bg-white shadow-[-1px_-0.5px_6px_rgba(0,0,0,0.07),1px_4px_6px_rgba(0,0,0,0.07)]"
    data-testid="Faq-Item-Container"
  >
    <button
      class="w-full px-4 sm:px-6 py-6 sm:py-8 text-left font-semibold text-base sm:text-lg text-iw-title flex justify-between items-center hover:bg-iw-faqAccordionHover hover:rounded-t-md focus:outline-none transition-colors"
      data-testid="Faq-Item-Title-Button"
    >
      <span
        data-testid="Faq-Item-Title-Text"
      >
        FaqTitle
      </span>
      <span
        class="ml-2 sm:ml-4"
      >
        <svg
          class="sm:size-[20px]"
          color="#04051D"
          data-testid="Faq-Item-UpArrow"
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(4, 5, 29);"
          viewBox="0 0 512 512"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 217.9L383 345c9.4 9.4 24.6 9.4 33.9 0 9.4-9.4 9.3-24.6 0-34L273 167c-9.1-9.1-23.7-9.3-33.1-.7L95 310.9c-4.7 4.7-7 10.9-7 17s2.3 12.3 7 17c9.4 9.4 24.6 9.4 33.9 0l127.1-127z"
          />
        </svg>
      </span>
    </button>
    <div
      class="bg-iw-background rounded-b-md"
    >
      <div
        class="px-4 sm:px-6"
      >
        <div
          class="border-t border-gray-200 py-5 sm:py-6 text-sm sm:text-base text-iw-title leading-relaxed"
          data-testid="faq-item-content-text"
        >
          <p
            class="mb-3 last:mb-0"
          >
            FaqContent
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
